modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/PainlessParser.java

```diff
@@ -2834,8 +2837,21 @@ public final LambdaContext lambda() throws RecognitionException {
       }
       setState(448);
       match(ARROW);
-      setState(449);
-      block();
+      setState(451);
+      switch ( getInterpreter().adaptivePredict(_input,38,_ctx) ) {
```

Comment:
@nik9000 On the brace issue -- I haven't had much time to investigate this.  So if you wanted to take a look it would be much appreciated.  I guess to get this bug to work now the lambda would have to use a block.